src/tokens/tokens.service.ts

```diff
@@ -0,0 +1,217 @@
+import { HttpService, Injectable, Logger } from '@nestjs/common';
+import { EventListener } from '../eventstream-proxy/eventstream-proxy.interfaces';
+import { EventStreamProxyGateway } from '../eventstream-proxy/eventstream-proxy.gateway';
+import { isFungible, packTokenId, packTokenUri, unpackTokenId, unpackTokenUri } from '../util';
+import { Event, EventStreamReply } from '../event-stream/event-stream.interfaces';
+import {
+  AsyncResponse,
+  EthConnectAsyncResponse,
+  EthConnectReturn,
+  ReceiptEvent,
+  TokenBalance,
+  TokenBalanceQuery,
+  TokenMint,
+  TokenMintEvent,
+  TokenPool,
+  TokenPoolEvent,
+  TokenTransfer,
+  TokenTransferEvent,
+  TokenType,
+  TransferSingleEventData,
+  UriEventData,
+} from './tokens.interfaces';
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+const uriEventSignature = 'URI(string,uint256)';
+const transferSingleEventSignature = 'TransferSingle(address,address,address,uint256,uint256)';
+
+@Injectable()
+export class TokensService {
+  baseUrl: string;
+  instanceUrl: string;
+  identity: string;
+  listener: TokenListener;
+
+  constructor(private http: HttpService, proxy: EventStreamProxyGateway) {
+    this.listener = new TokenListener(proxy);
+    proxy.addListener(this.listener);
+  }
+
+  configure(baseUrl: string, instanceUrl: string, identity: string) {
+    this.baseUrl = baseUrl;
+    this.instanceUrl = instanceUrl;
+    this.identity = identity;
+  }
+
+  private get postOptions() {
+    return {
+      params: {
+        'fly-from': this.identity,
```

Comment:
The short prefix here probably needs to be configurable, rather than relying on it being `fly-` in all cases (it could be `kld-` for example in a Kaleido hosted environment with ethconnect)